test/integration/lib/foxboxHelper.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const Config = require('config-js');
+const spawn = require('child_process').spawn;
+
+var config = new Config('./test/integration/lib/config/foxbox.js');
+var FOXBOX_STARTUP_WAIT_TIME_IN_MS = 3000;
+var foxboxInstance;
+
+var helper = (function() {
+
+  function removeUsersDB() {
+    var filePath = path.join(process.env.HOME, 
+      '/.local/share/foxbox/users_db.sqlite'); 
+    fs.unlinkSync(filePath);
+  }
+
+  function fullOptionStart(callback) {
+    
+    var start = new Promise(function (resolve, reject) {
+      foxboxInstance = spawn('./target/debug/foxbox', 
+       ['-c',  config.get('nupnp_server.param')+';'+ 
+       config.get('nupnp_server.url')+':'+ 
+       config.get('nupnp_server.port')+'/',
+       '-r',config.get('pagekite.r'),
+       '-t',config.get('pagekite.t'),
+       '-s',config.get('pagekite.s'),
+       '--remote-name',config.get('pagekite.externalURL'),
+       '--disable-tls'], {stdio: 'inherit'} ); // TODO TLS not yet supported
+      setTimeout(function(){resolve();}, FOXBOX_STARTUP_WAIT_TIME_IN_MS);
+    });
+     start.then(function(val) {return callback();}); 
```

Comment:
so callback has to be in this location.  if done() is called within above promise, it executes instantly, if there is a way around it, please let me know.